### PR TITLE
feat: add installation, first deployment steps

### PR DESCRIPTION
**Pull request by randomicecube**
@Pesteves2002 Currently missing verification in the deployment step + need to check if the `piku-nginx` needs to be active

### CODE REVIEW

File: tutorial/first-deploy/first-deploy.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#### **Lifecycle of a Piku Deployment**
+
+Here’s an internal flow of what happens when you push your code to Piku:
+1. `git push`: You push your app code to the Piku server using Git. The server has a Git remote repository set up to handle these pushes.
+   
+2. **Code Handling**: Upon receiving the push, Piku checks for a `Procfile` and `ENV` files. It also ensures any dependencies (e.g., Python, Node.js, etc.) are installed according to the language-specific packaging tools (like `requirements.txt` for Python).
+
+3. **Process Setup**: Based on the `Procfile`, Piku sets up process definitions using `systemd` or `supervisord`, creating background services for your app.
+   
+4. **Environment Injection**: Piku injects any environment variables from the `.env` file into the running processes, ensuring they have the necessary configuration to function.
```
Comment (Pesteves2002) [inline tutorial/first-deploy/first-deploy.md]:
```suggestion
4. **Environment Injection**: Piku injects any environment variables from the `ENV` file into the running processes, ensuring they have the necessary configuration to function.
```

**Review by Pesteves2002 — CHANGES_REQUESTED** [review]

**Comment by randomicecube** [comment]
TODO: change the mention to multiple hosts not being possible; either use multiple hosts, or simply don't mention it
The first one is probably best

**Comment by randomicecube** [comment]
@Pesteves2002 what do you think re: last commit?

File: tutorial/first-deploy/first-deploy.md [hunk 2]
```diff
@@ -67,6 +67,12 @@ git clone https://github.com/piku/sample-nodejs-app
 cd sample-nodejs-app
 ```{{exec}}
 
+We'll want to be able to deterministically check whether or not the application has been correctly deployed. To do this, we'll set up a port for the application to run on (using the `PORT` environment variable):
+
+```bash
+echo "PORT=2482" > ENV
```
Comment (Pesteves2002) [inline tutorial/first-deploy/first-deploy.md]: > replace with `echo "PORT=2482" >> ENV`

Comment (randomicecube) [reply tutorial/first-deploy/first-deploy.md]: oops, fair

File: tutorial/first-deploy/first-deploy.md [hunk 3]
```diff
@@ -67,6 +67,12 @@ git clone https://github.com/piku/sample-nodejs-app
...
+```{{exec}}
+
 ## Deploying
```
Comment (Pesteves2002) [inline tutorial/first-deploy/first-deploy.md:76]: Before this step you need to commit the changes

**Review by Pesteves2002 — CHANGES_REQUESTED** [review]

**Pull request merged**